### PR TITLE
destination is set up for API calls

### LINKED ISSUE #50: add Geocoder API
Adding the GeoApify API to the back end to take in a city name and return LAT/LONG coordinates. Those will then be passed into the Leaflet API to render on a map.

### PR DESCRIPTION
**Pull request by nicolepeak**
I found a json parser to be able to sift through the json that the GeoApify API returns and was able to pull out the lat and lon from that json to use in the leaflet.js in the front end. (functionality of this is in GeocodingService).

### CODE REVIEW

**Comment by nicolepeak** [comment]
I forgot to mention that I can to comment out all of the authentication code to bypass it. It is getting stuck in a loop redirecting to /login.

File: The-Coding-Sharks-api/src/main/java/com/example/The_Coding_Sharks_demo/controllers/DestinationController.java [hunk 1]
```diff
@@ -17,10 +18,24 @@ public class DestinationController {
...
+    private GeocodingService geocodingService;
+
     @PostMapping
-    public ResponseEntity<Destination> createDestination(@RequestBody Destination destination) {
-        Destination createdDestination = destinationService.saveDestination(destination);
-        return ResponseEntity.status(HttpStatus.CREATED).body(createdDestination); //201 response
+    public ResponseEntity<Destination> createDestination(@RequestParam String name) {
+
+        try {
+            System.out.println("TESTING");
```
Comment (adc009) [inline The-Coding-Sharks-api/src/main/java/com/example/The_Coding_Sharks_demo/controllers/DestinationController.java]: Let's remove this print statement before merging into main

File: The-Coding-Sharks-api/src/main/java/com/example/The_Coding_Sharks_demo/services/GeocodingService.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    private final HttpClient httpClient;
+    private final ObjectMapper objectMapper;
+
+    public GeocodingService() {
+        this.httpClient = HttpClient.newHttpClient();
+        this.objectMapper = new ObjectMapper();
+    }
+
+    public Destination geocodeDestination(String destinationName) throws Exception {
+        String apiKey = "3843f958ea2f4ba0beca427b6f821857";
```
Comment (adc009) [inline The-Coding-Sharks-api/src/main/java/com/example/The_Coding_Sharks_demo/services/GeocodingService.java:28]: We may want to revisit this later if we have time - this works fine now for our purposes but I think best practice is to not hardcode the API key.

**Review by adc009 — CHANGES_REQUESTED** [review]
Looks good! Minor nitpick - let's remove the print statement from DestinationController before pushing to main.

File: The-Coding-Sharks-api/src/main/java/com/example/The_Coding_Sharks_demo/controllers/DestinationController.java (hunk 1 above)
Comment (nicolepeak) [reply The-Coding-Sharks-api/src/main/java/com/example/The_Coding_Sharks_demo/controllers/DestinationController.java]: removed! thank you!

File: The-Coding-Sharks-api/src/main/java/com/example/The_Coding_Sharks_demo/services/GeocodingService.java (hunk 2 above)
Comment (nicolepeak) [reply The-Coding-Sharks-api/src/main/java/com/example/The_Coding_Sharks_demo/services/GeocodingService.java:28]: ok. I didn't realize that, but it totally makes sense. I will create another issue for this.

**Review by adc009 — APPROVED** [review]
Looks good!

**Pull request merged**